### PR TITLE
DAOS-11482 client, object, tests, utils: multiple coverity issues

### PR DESCRIPTION
**Pull request by wiliamhuang**
105914 Missing unlock
21957 Unchecked return value
21728 Unchecked return value
and several similar issues.

Required-githooks: true

Signed-off-by: Lei Huang <lei.huang@intel.com>

### CODE REVIEW

**Comment by wiliamhuang** [comment]
All CI tests passed.

File: src/tests/simple_array.c [hunk 1]
```diff
@@ -521,10 +521,14 @@ main(int argc, char **argv)
...
 
 	/** close container */
-	daos_cont_close(coh, NULL);
+	rc = daos_cont_close(coh, NULL);
+	if (rc)
+		D_ERROR("daos_cont_close() Failed "DF_RC"\n", DP_RC(rc));
 
 	/** disconnect from pool & destroy it */
-	daos_pool_disconnect(poh, NULL);
+	rc = daos_pool_disconnect(poh, NULL);
```
Comment (wangshilong) [inline src/tests/simple_array.c:529]: Not related to this PR, but looks rc might be lost and return 0.

**Review by wangshilong — DISMISSED** [review]

**Comment by wiliamhuang** [comment]
@daltonbohning Could you please review? Thank you very much!

File: src/object/cli_obj.c [hunk 2]
```diff
@@ -568,9 +568,10 @@ obj_ec_leader_select(struct dc_object *obj, int grp_idx, bool cond_modify, uint3
 	 * If no parity node is available, then handle related task that has conditional
 	 * modification via distributed tranasaction.
 	 */
-	if (cond_modify)
+	if (cond_modify) {
+		D_RWLOCK_UNLOCK(&obj->cob_lock);
 		return -DER_NEED_TX;
-
+	}
```
Comment (daltonbohning) [inline src/object/cli_obj.c:574]: Do we want to use `D_GOTO(unlock, -DER_NEED_TX)` to be consistent in this function?

**Review by daltonbohning — APPROVED** [review]

File: src/object/cli_obj.c (hunk 2 above)
Comment (wiliamhuang) [reply src/object/cli_obj.c]: Thank you! Just updated as you suggested.

**Review by daltonbohning — APPROVED** [review]

**Review by wangshilong — APPROVED** [review]

**Review by jolivier23 — APPROVED** [review]

**Comment by jolivier23** [comment]
why did you skip functional hardware?

**Comment by wiliamhuang** [comment]
> why did you skip functional hardware?

Hardware tests all passed in build 6 previously. Very minor change was made since then. Only one return was changed to goto for style consistency as Dalton suggested. Hardware tests were skipped to save CI resources.

**Comment by wiliamhuang** [comment]
@jolivier23 Can you land this PR? Or do you want me to run hardware tests again? Thank you!

**Comment by wiliamhuang** [comment]
@jolivier23 I merged with master and ran this PR again. All CI tests including hardware tests passed. I just removed "forced-landing" label. Could you please land this PR now? Thank you!

**Pull request merged**